File: Programmers/level2/carpet/carpet.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/42842

def solution(brown, yellow):
    answer = []
    total = brown + yellow
    for i in range(1, total+1):
        if total % i == 0:
            x = total // i
            y = i
            if x < y:
                break
            else:
                answer.append([x, y])
    # print(answer)

    for i in answer:
        if i[0] * 2 + i[1] * 2 - 4 == brown:
            return i

File: Programmers/level2/carpet/carpet.js
export const solution = (brown, yellow) => {
    let answer = [];
    let total = brown + yellow;
    for(let i = 1; i < total+1; i++){
        if(total % i === 0){
            let x = total / i;
            let y = i;
            if(x < y){
                break;
            } else {
                answer.push([x,y])
            }
        }
    }
    // console.log(answer)
    for(let i = 0; i < answer.length; i++){
        if(answer[i][0] * 2 + answer[i][1] * 2 - 4 === brown){
            return answer[i];
        }
    }
}